internal/controllers/addon_utils.go

```diff
@@ -96,6 +96,7 @@ func (r *AddonReconciler) parseAddonInstallConfig(
 		catalogSourceImage = addon.Spec.Install.AllNamespaces.CatalogSourceImage
 
 	default:
+		// TODO safe to remove this as kubernetes does schema validation??
```

Comment:
The extra schema validation, like enum will be validated when an object is created or updated, but (and I have to double check this) I don't think it will force-reset values that are already stored (it will definetly drop no-longer-existing fields and mismatched datatypes on load though).

Having a value in here that is not allowed by validation could happen during development with stale data still on the cluster or when we screw up updating our schema. Finding that error will be way easier when we have a log message and not just silently ignore this case.

So while we can provide a better user experience with validation and defaulting, our code can't rely 100% on read operations returning compliant data. We need to ensure that these cases will not crash - nil pointers and such, we report these unexpected errors and ideally have the system fix itself. :)

Edit: We should update the description of this block to explain the ^ reasons better.

Reply:
Interesting, so what you are saying is if we ever migrate our schema, there's a chance we will have legacy values stored in etcd, and then there is no Defaulting/Validating webhook or OpenAPI validation that can detect this. Only the controller can find those glitches and log them.
Let's keep the default case :)